src/main/config/feature.c

```diff
@@ -84,6 +84,13 @@ void featureDisableImmediate(const uint32_t mask)
     featureClear(mask, &runtimeFeatureMask);
 }
 
+// Updates *ONLY* the runtime state of a feature and not the configuration
+// Used *ONLY* by the config check process that runs at startup.
+void featureDisableRuntime(const uint32_t mask)
+{
+    featureClear(mask, &runtimeFeatureMask);
```

Comment:
One problem I can see with this is that features changed with this will still show as 'enabled' in configurator, even though they are not enabled, giving users an false and misleading representation of what the actual configuration is, and making it impossible to diagnose this.
If we want to use this we should change `MSP_FEATURE_CONFIG` to return `runtimeFeatureMask`.

Reply:
The key point is that having a feature "enabled" doesn't mean the feature is "working". Rather that the user is trying to get the feature to work. Take for example GPS. The feature can be enabled but there are lots of reasons why it might not be working. This is no different. The logic here is simply disabling the OSD functionality when it's not (currently) workable/needed in this power cycle. And as I mentioned in the comments, having the code do a hard-disable of the feature makes the UX very painful as there are multiple steps involving save/reboot to actually bet MSP DisplayPort OSD working and each step was turning the feature off. That's the real reason for this change - to avoid the end-user frustration about why the OSD feature keeps turning off. Further the MSP DisplayPort is a "push" type feature where we get no feedback as to whether anything is actually connected and working anyway. So even if everything is set up properly it's still possible that the feature isn't working even though it's enabled and the possible user confusion is unavoidable in this case. To keep beating a dead horse, if we followed the logic to disable features when not working, then why not turn off `RX_SERIAL` whenever we detect no radio signal at boot?

Reply:
Yes, GPS detection is a good example of horrible usability - diagnosing this is utterly frustrating and leading to users opening issues because the behaviour is not intuitive.
Not sure if disabling `RX_SERIAL` because the flight controller is plugged in via USB and the RX is not powered is going to be user friendly, but feel free to open a pull request for such a change if you feel it is needed.